src/mapbuffer.cpp

```diff
@@ -476,10 +507,34 @@ void mapbuffer::deserialize( JsonIn &jsin )
                         }
                     }
                 } else {
+                    // terrain is encoded using simple RLE
                     for( int j = 0; j < SEEY; j++ ) {
-                        for( int i = 0; i < SEEX; i++ ) {
```

Comment:
How about
```
int remaining_same = 0;
for(j)
    for(i)
        if(remainining_same > 0)
            remaining_same--
        else
            if is string
                iid = Decode string
            else if is aray
                iid, remaining_same = Decode array
                remaining_same--
        sm->ter[i][j] = iid;
```

Reply:
A huge improvement over the fever dream I managed to cook up.  Thanks.